lab_3/main.py

```diff
@@ -222,7 +338,23 @@ def save(self, name: str) -> int:
         :param name: name of the json file with .json format
         :return: 0 if profile saves, 1 if any errors occurred
         """
-        pass
+        if not isinstance(name, str):
+            return 1
+        profile_as_dict = {}
+        freq_dict = {}
+        new_key = ''
+        for trie in self.tries:
+            for key, value in trie.n_gram_frequencies.items():
+                for element in key:
+                    new_key += self.storage.get_letter_by_id(element)
+                freq_dict[new_key] = value
+                new_key = ''
+        profile_as_dict['n_words'] = self.n_words
+        profile_as_dict['name'] = self.language
+        with open('{}.json'.format(name), 'w', encoding="utf-8") as file:
```

Comment:
`file` is the Python built-in. Consider using another name.

Reply:
Done